src/neptune/new/internal/disk_queue.py

```diff
@@ -152,6 +153,22 @@ def close(self):
         self._last_ack_file.close()
         self._last_put_file.close()
 
+        if self.is_empty():
+            self._remove_data()
+
+    def _remove_data(self):
+        path = self._dir_path
+        shutil.rmtree(path)
```

Comment:
Not sure but think of exception that it may raise or even think about `ignore_errors=False`